### ISSUE TITLE
dns isn't connecting to wfc

### ISSUE THREAD

**Issue by ryrydalion** [kuroppoi/entralinked#17]
i've done everything up until having to connect my game to the program but, the dns it's giving me doesn't connect to nintendo wfc. i also tried to do the update pid and it keeps saying my wifi connection doesn't exist.

**Comment by kuroppoi** [comment]
What error code do you get?

**Comment by ryrydalion** [comment]
> What error code do you get?

when i try and game sync i get 20100 "unable to connect to nintendo wifi connection"

**Comment by kuroppoi** [comment]
Could you post Entralinked’s log here?

**Comment by ryrydalion** [comment]
> Could you post Entralinked’s log here?

like a photo of the program on my computer and do i show the dns or block it out

**Comment by ryrydalion** [comment]
2023-09-13 12:27:04.021 ERROR : Could not add BouncyCastle to SystemClassLoader search
java.nio.file.NoSuchFileException: libs\bcutil-jdk15on-1.70.jar
	at sun.nio.fs.WindowsException.translateToIOException(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsFileSystemProvider.newByteChannel(Unknown Source) ~[?:?]
	at java.nio.file.spi.FileSystemProvider.newOutputStream(Unknown Source) ~[?:?]
	at java.nio.file.Files.newOutputStream(Unknown Source) ~[?:?]
	at java.nio.file.Files.copy(Unknown Source) ~[?:?]
	at entralinked.LauncherAgent.agentmain(LauncherAgent.java:35) ~[entralinked.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(Unknown Source) ~[?:?]
	at sun.instrument.InstrumentationImpl.loadClassAndCallAgentmain(Unknown Source) ~[?:?]
	at sun.instrument.InstrumentationImpl.loadAgent0(Native Method) ~[?:?]
	at sun.instrument.InstrumentationImpl.loadAgent(Unknown Source) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at sun.launcher.LauncherHelper.lambda$getMainClassFromJar$0(Unknown Source) ~[?:?]
	at java.util.Optional.ifPresent(Unknown Source) ~[?:?]
	at sun.launcher.LauncherHelper.getMainClassFromJar(Unknown Source) ~[?:?]
	at sun.launcher.LauncherHelper.loadMainClass(Unknown Source) ~[?:?]
	at sun.launcher.LauncherHelper.checkAndLoadMain(Unknown Source) ~[?:?]
2023-09-13 12:27:04.036  INFO : Loading configuration ...
2023-09-13 12:27:04.036  INFO : No configuration file exists - default configuration will be used
2023-09-13 12:27:04.037 ERROR : Could not load configuration - default configuration will be used
java.io.FileNotFoundException: config.json (Access is denied)
	at java.io.FileOutputStream.open0(Native Method) ~[?:?]
	at java.io.FileOutputStream.open(Unknown Source) ~[?:?]
	at java.io.FileOutputStream.<init>(Unknown Source) ~[?:?]
	at java.io.FileOutputStream.<init>(Unknown Source) ~[?:?]
	at com.fasterxml.jackson.core.TokenStreamFactory._fileOutputStream(TokenStreamFactory.java:294) ~[entralinked.jar:?]
	at com.fasterxml.jackson.core.JsonFactory.createGenerator(JsonFactory.java:1428) ~[entralinked.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper.createGenerator(ObjectMapper.java:1230) ~[entralinked.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:3906) ~[entralinked.jar:?]
	at entralinked.Entralinked.loadConfigFile(Entralinked.java:174) [entralinked.jar:?]
	at entralinked.Entralinked.<init>(Entralinked.java:69) [entralinked.jar:?]
	at entralinked.Entralinked.main(Entralinked.java:33) [entralinked.jar:?]
2023-09-13 12:27:04.038  INFO : Using configuration Configuration[hostName=local, clearPlayerDreamInfoOnWake=true, allowOverwritingPlayerDreamInfo=false, allowPlayerGameVersionMismatch=false, allowWfcRegistrationThroughLogin=true]
2023-09-13 12:27:04.104  INFO : Loading DLC ...
2023-09-13 12:27:04.104  INFO : Extracting default DLC files ...
2023-09-13 12:27:04.107 ERROR : Could not extract default DLC files
java.nio.file.NoSuchFileException: .\dlc\IRAO\CGEAR\01 - Default.bin
	at sun.nio.fs.WindowsException.translateToIOException(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsFileSystemProvider.newByteChannel(Unknown Source) ~[?:?]
	at java.nio.file.spi.FileSystemProvider.newOutputStream(Unknown Source) ~[?:?]
	at java.nio.file.Files.newOutputStream(Unknown Source) ~[?:?]
	at java.nio.file.Files.copy(Unknown Source) ~[?:?]
	at entralinked.model.dlc.DlcList.<init>(DlcList.java:48) [entralinked.jar:?]
	at entralinked.Entralinked.<init>(Entralinked.java:94) [entralinked.jar:?]
	at entralinked.Entralinked.main(Entralinked.java:33) [entralinked.jar:?]
2023-09-13 12:27:04.107  INFO : Loading user and profile data ...
2023-09-13 12:27:04.108  INFO : Loading player data ...
2023-09-13 12:27:04.229  INFO : Mapping GameSpy request types ...
2023-09-13 12:27:04.255  WARN : SSL will be disabled because BouncyCastle could not be loaded
2023-09-13 12:27:04.362  INFO : Loading C-Gear and Pokédex skin previews ...
2023-09-13 12:27:04.363  INFO : Cached 0 skin previews
2023-09-13 12:27:04.367  INFO : Starting servers ...
2023-09-13 12:27:04.368  INFO : Starting HTTP server ...
2023-09-13 12:27:04.369  INFO : Starting Javalin ...
2023-09-13 12:27:04.394  INFO : jetty-11.0.15; built: 2023-04-11T18:37:53.775Z; git: 5bc5e562c8d05c5862505aebe5cf83a61bdbcb96; jvm 17.0.8.1+1
2023-09-13 12:27:04.434  INFO : Session workerName=node0
2023-09-13 12:27:04.444  INFO : Started i.j.j.@3aefae67{/,null,AVAILABLE}
2023-09-13 12:27:04.455  INFO : Started ServerConnector@db44aa2{HTTP/1.1, (http/1.1)}{0.0.0.0:80}
2023-09-13 12:27:04.456  INFO : Started Server@33a053d{STARTING}[11.0.15,sto=0] @1285ms
2023-09-13 12:27:04.456  INFO : 
       __                  ___          ______
      / /___ __   ______ _/ (_)___     / ____/
 __  / / __ `/ | / / __ `/ / / __ \   /___ \
/ /_/ / /_/ /| |/ / /_/ / / / / / /  ____/ /
\____/\__,_/ |___/\__,_/_/_/_/ /_/  /_____/

       https://javalin.io/documentation

2023-09-13 12:27:04.472  INFO : Static file handler added: StaticFileConfig(hostedPath=/dashboard, directory=/dashboard, location=CLASSPATH, precompress=false, aliasCheck=null, headers={Cache-Control=max-age=0}, skipFileFunction=Function1<jakarta.servlet.http.HttpServletRequest, java.lang.Boolean>, mimeTypes=io.javalin.http.staticfiles.MimeTypesConfig@3703bf3c). File system location: 'jar:file:/C:/Users/ryan/Downloads/entralinked/entralinked.jar!/dashboard/'
2023-09-13 12:27:04.473  INFO : Static file handler added: StaticFileConfig(hostedPath=/sprites, directory=/sprites, location=CLASSPATH, precompress=false, aliasCheck=null, headers={Cache-Control=max-age=0}, skipFileFunction=Function1<jakarta.servlet.http.HttpServletRequest, java.lang.Boolean>, mimeTypes=io.javalin.http.staticfiles.MimeTypesConfig@1dab9dd6). File system location: 'jar:file:/C:/Users/ryan/Downloads/entralinked/entralinked.jar!/sprites/'
2023-09-13 12:27:04.475  INFO : Listening on http://localhost:80/
2023-09-13 12:27:04.483  INFO : You are running Javalin 5.5.0 (released May 1, 2023. Your Javalin version is 134 days old. Consider checking for a newer version.).
2023-09-13 12:27:04.484  INFO : Javalin started in 116ms \o/
2023-09-13 12:27:04.486  INFO : Staring GameSpy server ...
2023-09-13 12:27:04.566  INFO : GameSpy server listening @ port 29900
2023-09-13 12:27:04.567  INFO : Staring DNS server ...
2023-09-13 12:27:04.578  INFO : DNS server listening @ port 53
2023-09-13 12:27:04.579  INFO : Startup complete! Took a total of 984 milliseconds

**Comment by ryrydalion** [comment]
> The issue is that SSL is a disabled because critical dependencies could not be extracted because the program has no write permissions. Without SSL, your DS refuses to connect.
> Try moving `entralinked.jar` out of your `Downloads` folder.

i just took it out of my downloads folder and put it on my desktop and it still says ssl will be disabled because bouncycastle could not be loaded

**Comment by kuroppoi** [comment]
I guess something is preventing Java from creating files. I don't know what or why, maybe an antivirus of some kind.
Try running a Command Prompt as administrator and execute the command `java -jar entralinked.jar`

**Comment by ryrydalion** [comment]
i got this: C:\Users\ryan>java -jar entralinked.jar
Error: Unable to access jarfile entralinked.jar

**Comment by kuroppoi** [comment]
Yeah you have to navigate to wherever `entralinked.jar` is first.
If you put `entralinked.jar` in a folder called `entralinked` on your desktop, you can do
`cd desktop/entralinked`
and then
`java -jar entralinked.jar`

**Comment by ryrydalion** [comment]
it worked like a charm let me try again now

**Comment by ryrydalion** [comment]
thank you so much yo it worked
